### PR TITLE
reftest: Escape regexps characters in string replacements primitives

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Alternative to https://github.com/ocaml/opam/pull/5007

### CODE REVIEW

**Comment by AltGr** [comment]
Doesn't need to b an alternative, I think we want both

**Comment by AltGr** [comment]
(this patch is a very good idea and much more general; but the opamroot-version test remains brittle even with this, you turn a 2-char match to a 3-char one which is still pretty short)

File: tests/reftests/run.ml [hunk 1]
```diff
@@ -136,17 +136,22 @@ type filt_sort =
   | Grep
   | GrepV
 
+let escape_all s =
+  let buf = Buffer.create (String.length s * 2) in
+  String.iter (fun c -> Buffer.add_char buf '\\'; Buffer.add_char buf c) s;
+  Buffer.contents buf
```
[the remark this answers is not in the document]
Comment (kit-ty-kate) [reply tests/reftests/run.ml:142]: mmh, fair point, the POSIX standard says it’s undefined and ocaml-re does fail. Gimme one second to fix it

[the remark this answers is not in the document]
Comment (AltGr) [reply tests/reftests/run.ml:142]: hm actually with `Re.Posix.re "\\a"` I even get a `Re__Posix.Parse_error` exception :(

**Comment by rjbou** [comment]
Thanks!

**Pull request merged**